hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/repair/ldb/TestLdbRepair.java

```diff
@@ -0,0 +1,153 @@
+  private ManagedDBOptions options;
+  
+  @BeforeEach
+  public void setUp() throws Exception {
+    CodecBuffer.enableLeakDetection();
+    
+    dbPath = tempDir.resolve("test.db");
+    options = new ManagedDBOptions();
+    options.setCreateIfMissing(true)
+        .setCreateMissingColumnFamilies(true);
+    rdbStore = DBStoreBuilder.newBuilder(new OzoneConfiguration())
+        .setName(dbPath.toFile().getName())
+        .setPath(dbPath.getParent())
+        .addTable(TEST_CF_NAME)
+        .build();
+  }
+  
+  @AfterEach
+  public void tearDown() throws Exception {
+    if (rdbStore != null && !rdbStore.isClosed()) {
+      rdbStore.close();
+    }
+    if (options != null) {
+      options.close();
+    }
+    CodecBuffer.assertNoLeaks();
+  }
+
+  /**
+   * Test manual compaction of RocksDB.
+   * This test creates a large number of keys, deletes them, and then triggers compaction.
+   * The sizes after each step is compared to ensure that compaction reduces the size of the database.
+   */
+  @Test
+  public void testRocksDBManualCompaction() throws Exception {
+    Table<byte[], byte[]> testTable = rdbStore.getTable(TEST_CF_NAME);
+
+    // Create many keys
+    for (int i = 0; i < NUM_KEYS; i++) {
+      String key = "key" + i;
+      String value = RandomStringUtils.secure().nextAlphanumeric(100);
+      testTable.put(key.getBytes(StandardCharsets.UTF_8), value.getBytes(StandardCharsets.UTF_8));
+    }
+    rdbStore.flushDB();
+    long size1 = calculateSstFileSize(dbPath);
+    
+    // Delete all keys
+    for (int i = 0; i < NUM_KEYS; i++) {
+      String key = "key" + i;
+      testTable.delete(key.getBytes(StandardCharsets.UTF_8));
+    }
+    rdbStore.flushDB();
+    long size2 = calculateSstFileSize(dbPath);
+    rdbStore.close();
+
+    // Trigger compaction of the table
+    RocksDBManualCompaction compactionTool = new RocksDBManualCompaction();
+    CommandLine cmd = new CommandLine(compactionTool);
+    String[] args = {
+        "--db", dbPath.toString(),
+        "--column-family", TEST_CF_NAME
+    };
+    int exitCode = withTextFromSystemIn("y")
+        .execute(() -> cmd.execute(args));
+    assertEquals(0, exitCode, "Compaction command should execute successfully");
+    long size3 = calculateSstFileSize(dbPath);
+    
+    System.out.println("Size after adding keys (size1): " + size1);
+    System.out.println("Size after deleting keys (size2): " + size2);
+    System.out.println("Size after compaction (size3): " + size3);
+    
+    // size1 < size2 as deletes increase size due to tombstones
+    assertTrue(size1 < size2, 
+        String.format("Expected size1 (%d) < size2 (%d), but size1 >= size2", size1, size2));
+    
+    // size3 < size1 as compaction should reduce size below original
+    assertTrue(size3 < size1,
+        String.format("Expected size3 (%d) < size1 (%d), but size3 >= size1", size3, size1));
+  }
+
+  private long calculateSstFileSize(Path db) throws IOException {
```

Comment:
How about using the `RdbUtil#getLiveSstFilesForCFs` to get the `LiveMetadata` and then ensuring that the `numDeletions` are 0.